src/store.rs

```diff
@@ -478,7 +478,7 @@ impl Core {
     }
 
     fn append_log(&self, tx_record: &BytesMut, durability: Durability) -> Result<(u64, u64)> {
-        let mut clog = self.clog.as_ref().unwrap().write();
+        let clog = self.clog.as_ref().unwrap().write();
```

Comment:
With `Aol` made thread safe using `Mutex` internally, I think we might be able do drop the `RwLock` protecting `clog' at the `Store` level, because it will serve as the serialisation mechanism between `close()` and `append_log()` as stated in https://github.com/surrealdb/surrealkv/pull/176/files#diff-41a991785a2eba02345db5fdcbf15b660f27b2841b8f61a084a7db66e69bf3eeL413-L421

Reply:
I agree, I thought of making/testing those changes in a separate PR so that any anomalies introduced through this code can then be isolated